packages/react-dev-utils/BlockUnsupportedWebpackLoaderSyntax.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+// This Webpack plugin forbids usage of special webpack syntax
+// to disable loaders (https://webpack.github.io/docs/loaders.html#loader-order)
+// This makes it very coupled to webpack and might break in the future.
+// See https://github.com/facebookincubator/create-react-app/issues/733.
+
+'use strict';
+
+class BlockUnsupportedWebpackLoaderSyntax {
+  apply(compiler) {
+    compiler.plugin('normal-module-factory', (normalModuleFactory) => {
+      normalModuleFactory.plugin('before-resolve', (data, callback) => {
+        let error = null;
+        if (data.request.match(/^-?!!?/)) {
```

Comment:
It'd be best to use the exact same regexps as webpack uses for this (which I believe can be found here https://github.com/webpack/webpack/blob/master/lib/NormalModuleFactory.js#L86)